### ISSUE TITLE
Go to Definition on includes can get stuck due to hover or documentHighlight requests that are stuck on IntelliSense updating (from document switching)

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#8885]
### Discussed in https://github.com/microsoft/vscode-cpptools/discussions/8881

<div type='discussions-op-text'>

<sup>Originally posted by **juxeii** February 18, 2022</sup>
We have a very large code base.
One problem I have with Intellisense is that it reparses the active document although it has been parsed before.
Here is an example of how I cannot work in a smooth fashion:
I mark one include file A and click "Go to Definition".
The file A opens in an instant which is fine.
Now I go back to the original file from where I jumped from and browse to other include files.
Now I want to browse to file A again(no changes were made). It can take many seconds now to open the file A again.

I wonder why is that? Why is there no cache used which opens the file A again in an instant?
On the console for C/C++ I can see "idle loop: reparsing the active document".
Is there something I can do to improve the user experience here?
</div>

**Comment by juxeii** [comment]
@sean-mcmanus: Yes, I noticed there are not logger outputs while it tries to jump to the file in question.

**Comment by Colengms** [comment]
@sean-mcmanus   A hang in documentHighlight could be https://github.com/microsoft/vscode-cpptools/issues/8717

**Comment by sean-mcmanus** [comment]
@Colengms No, I'm just referring to a normal delay of document highlight due to it being stuck waiting for the IntelliSense update.

**Comment by simonboily** [comment]
Has there been any progress on this issue? Any work arounds?  This happens even on relatively small projects, making VSCode a no-go tool for me.

**Comment by sean-mcmanus** [comment]
@simonboily Sorry, no progress yet. If you can avoid hovering or other mouse movement and do Go to Def via F12 that could help.

**Comment by Kerdek** [comment]
An extension can assume the responsibility of avoiding reparsing the document by storing the TextDocument reference in a map used to look up the parse result.

I recommend wrapping this behavior and calling it a day.
